apps/confidential/confidential.go

```diff
@@ -156,11 +156,20 @@ func NewCredFromSecret(secret string) (Credential, error) {
 }
 
 // NewCredFromAssertion creates a Credential from a signed assertion.
+//
+// Deprecated: a Credential created by this function can't refresh the
+// assertion when it expires. Use NewCredFromAssertionCallback instead.
 func NewCredFromAssertion(assertion string) (Credential, error) {
 	if assertion == "" {
 		return Credential{}, errors.New("assertion can't be empty string")
 	}
-	return Credential{assertion: assertion}, nil
+	return NewCredFromAssertionCallback(func(context.Context) (string, error) { return assertion, nil }), nil
+}
+
+// NewCredFromAssertionCallback creates a Credential that invokes a callback to get assertions
+// authenticating the application.
```

Comment:
should we document that this needs to be thread safe?

Reply:
Yes, thanks for suggesting it.